### PR TITLE
Fix conflicts on infinite scroll

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
fixes https://github.com/EveripediaNetwork/issues/issues/175

### CODE REVIEW

File: src/pages/categories/[category].tsx [hunk 1]
```diff
@@ -23,17 +25,41 @@ import WikiPreviewCard from '@/components/Wiki/WikiPreviewCard/WikiPreviewCard'
 import { getWikisByCategory } from '@/services/wikis'
 import { Wiki } from '@/types/Wiki'
 import { useRouter } from 'next/router'
+import InfiniteScroll from 'react-infinite-scroll-component'
 
-interface CategoryPageProps {
+const limit = 3
```
[the remark this answers is not in the document]
Comment (Emmanueldmlr) [reply src/pages/categories/[category].tsx:30]: Thanks for that. I will update it now.

File: src/pages/categories/[category].tsx [hunk 2]
```diff
@@ -23,17 +25,41 @@ import WikiPreviewCard from '@/components/Wiki/WikiPreviewCard/WikiPreviewCard'
...
+        if (result.data && result.data?.length > 0) {
+          const data = result.data || []
+          const updatedWiki = [...wikisInCategory, ...data]
+          setWikisInCategory(updatedWiki)
+        } else {
+          setHasMore(false)
+        }
+      }
+      fetchNewWikis()
+    }, 3000)
```
Comment (kesar) [inline src/pages/categories/[category].tsx:61]: move this to a constant near the prev. one

**Comment by kesar** [comment]
there is a bug when navigating between categories:

if you go to defi category and then click in navbar menu > nft categories, they dont get refreshed. It's not fully related to this PR but probably worth to fix it here at the same time w do the pagination.

File: src/pages/categories/[category].tsx [hunk 3]
```diff
@@ -23,17 +25,41 @@ import WikiPreviewCard from '@/components/Wiki/WikiPreviewCard/WikiPreviewCard'
 import { getWikisByCategory } from '@/services/wikis'
 import { Wiki } from '@/types/Wiki'
 import { useRouter } from 'next/router'
+import InfiniteScroll from 'react-infinite-scroll-component'
 
-interface CategoryPageProps {
+const limit = 3
+let offset = 0
```
[the remark this answers is not in the document]
Comment (Emmanueldmlr) [reply src/pages/categories/[category].tsx:31]: moving the offset to be part of the component state implies that getServerSideProps will not have access to the offset variable. The major reason why I had to move it out of the component.

**Review by kesar — APPROVED** [review]

**Comment by Emmanueldmlr** [comment]
@kesar, I have attended to all issues raised. you can test it using this link: https://monopedia-bb3t8bthx-prediqt.vercel.app/categories/events

**Pull request merged**